src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java

```diff
@@ -0,0 +1,291 @@
+            try {
+                submitGiftCodes();
+            } catch (MoneyTransferException e) {
+                try {
+                    sendNotification(e.toString(), true);
+                } catch (MoneyTransferException f) {
+                    LOGGER.error(f);
+                }
+                LOGGER.error(e);
+            }
+        };
+
+
+        schedule = scheduler.scheduleAtFixedRate(runnable, 7, 7, TimeUnit.DAYS);
+
+    }
+
+    /**
+     * Shuts the MoneyTransferManager down.
+     */
+    public synchronized void shutdown() {
+        schedule.cancel(false);
+        scheduler.shutdown();
+        schedule = null;
+    }
+
+    /**
+     * Logs a new money transfer and saves it.
+     *
+     * @param workerID the id of the worker, who gets the money
+     * @param amount   the amount of money in ct
+     */
+    public void addMoneyTransfer(int workerID, int amount, int expID) {
+        workerBalanceOperations.addCredit(workerID, amount, expID);
+    }
+
+    /**
+     * Pays all workers depending on their logged money transfers.
+     * @throws MoneyTransferException gets thrown, if an error occurred
+     */
+    public void submitGiftCodes() throws MoneyTransferException {
+        LOGGER.trace("Started fetching new giftcodes.");
+        fetchNewGiftCodes();
+        LOGGER.trace("Completed fetching new giftcodes.");
+
+        LOGGER.trace("Started submission of giftcodes to workers.");
+        Result<WorkerRecord> workers = workerOperations.getWorkerWithCreditBalanceGreaterOrEqual(payOffThreshold);
+        List<GiftCodeRecord> giftCodes = workerBalanceOperations.getUnusedGiftCodes();
+
+        for (WorkerRecord worker : workers) {
+            if (!worker.getEmail().equals("")) {
+                List<GiftCodeRecord> payedCodesForWorker = chooseGiftCodes(worker, giftCodes);
+
+                payWorker(worker, payedCodesForWorker);
+                giftCodes = workerBalanceOperations.getUnusedGiftCodes();
+            }
+        }
+        LOGGER.trace("Completed submission of giftcodes to workers.");
+
+        //sends a notification about problems with submission of giftcodes
+        sendNotification(notificationText.toString(), false);
+    }
+
+    /**
+     * Fetches new giftcodes from the mailbox and saves them in the database.
+     *
+     * @throws MoneyTransferException gets thrown, if an error occurred
+     */
+    private void fetchNewGiftCodes() throws MoneyTransferException {
+        Message[] messages;
+
+        //fetch new mails
+        try {
+            messages = mailHandler.fetchUnseen("inbox");
+        } catch (MessagingException e) {
+            throw new MoneyTransferException("The MailHandler couldn't fetch new giftcodes from the mailserver. " +
+                    "It seems, that there is either a problem with the server or with the properties file.");
+        }
+
+        //extract giftcodes and save them to the database
+        for (Message message : messages) {
```

Comment:
Please log all `messageIds` as `INFO`. Might also save them in the DB and keep our own `unseen` state instead of trusting the inbox.